pkg/reconciler/broker/broker_test.go

```diff
@@ -355,11 +356,93 @@ func TestAllCases(t *testing.T) {
 				},
 			}),
 		},
+	}, {
+		Name: "Create broker with ready brokercell with nil Pubsub client",
+		Key:  testKey,
+		Objects: []runtime.Object{
+			NewBroker(brokerName, testNS,
+				WithBrokerClass(brokerv1beta1.BrokerClass),
+				WithBrokerUID(testUID),
+				WithBrokerDeliverySpec(brokerDeliverySpec),
+				WithBrokerSetDefaults),
+			NewBrokerCell(resources.DefaultBrokerCellName, systemNS,
+				WithBrokerCellReady,
+				WithBrokerCellSetDefaults),
+		},
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: NewBroker(brokerName, testNS,
+				WithBrokerClass(brokerv1beta1.BrokerClass),
+				WithBrokerUID(testUID),
+				WithBrokerDeliverySpec(brokerDeliverySpec),
+				WithBrokerReadyURI(brokerAddress),
+				WithBrokerSetDefaults,
+			),
+		}},
+		WantEvents: []string{
+			brokerFinalizerUpdatedEvent,
+			Eventf(corev1.EventTypeNormal, "TopicCreated", `Created PubSub topic "cre-bkr_testnamespace_test-broker_abc123"`),
+			Eventf(corev1.EventTypeNormal, "SubscriptionCreated", `Created PubSub subscription "cre-bkr_testnamespace_test-broker_abc123"`),
+			brokerReconciledEvent,
+		},
+		WantPatches: []clientgotesting.PatchActionImpl{
+			patchFinalizers(testNS, brokerName, brokerFinalizerName),
+		},
+		OtherTestData: map[string]interface{}{
+			"pre":                   []PubsubAction{},
+			"maxPSClientCreateTime": 1,
```

Comment:
To confirm I understand, is this intended to check that we never attempt to create the client twice?

Reply:
Yes.